src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/share/utils/rule/validation/SimpleSwitchRuleConverter.java

```diff
@@ -114,9 +119,26 @@ private SimpleSwitchRule buildIngressSimpleSwitchRule(Flow flow, FlowPath flowPa
             } else if (flow.getEncapsulationType().equals(FlowEncapsulationType.VXLAN)) {
                 rule.setTunnelId(encapsulationId.getEncapsulationId());
             }
+
+            if (ingress.isLooped() && !flowPath.isProtected()) {
+                rules.add(buildIngressLoopSimpleSwitchRule(rule, flowPath, ingress));
+            }
         }
+        rules.add(rule);
```

Comment:
Here you have a direct and indirect fill of `rules`. I would like to not mix these approaches, it makes it difficult to understand from reading in one glance.

Reply:
Fixed.